src/GitHub.VisualStudio.UI/Helpers/ThemeDictionaryManager.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Windows;
+using System.ComponentModel;
+using Microsoft.VisualStudio.PlatformUI;
+using GitHub.VisualStudio.Helpers;
+using GitHub.Helpers;
+
+namespace GitHub.VisualStudio.UI.Helpers
+{
+    public class ThemeDictionaryManager : SharedDictionaryManagerBase, IDisposable
+    {
+        static bool isInDesignMode;
+        Uri baseThemeUri;
+
+        static ThemeDictionaryManager()
+        {
+            isInDesignMode = DesignerProperties.GetIsInDesignMode(new DependencyObject());
+        }
+
+        public override Uri Source
+        {
+            get { return base.Source; }
+            set
+            {
+                InitTheme(value);
+                base.Source = GetCurrentThemeUri();
+            }
+        }
+
+        public void Dispose()
+        {
+            if(baseThemeUri != null)
+            {
+                baseThemeUri = null;
+                if (!isInDesignMode)
+                {
+                    VSColorTheme.ThemeChanged -= OnThemeChange;
+                }
+            }
+        }
+
+        void InitTheme(Uri themeUri)
```

Comment:
Personally I'd rather just set a property so that I can see e.g. how my design looks in the dark theme without having to go to the options menu, set the theme and then change it back when I've finished...